lib/dslash_policy.cuh

```diff
@@ -435,7 +439,72 @@ namespace quda
     }
   };
 
-/**
+  /**
+     Generic shmem dslash
+      // shmem bitfield encodes
```

Comment:
Readability thought, we should make this an `enum` in `enum_quda.h` to remove the random magic numbers in the code.

Reply:
Yes, I was waiting for C++17 to clean that up using something like https://github.com/Neargye/magic_enum

Reply:
Fair, and I'm happy deferring it, but it'd be nice to clean up the random magic numbers sooner as opposed to later in my opinion, even though abusing enum <-> int casts can be headache inducing.